anker/src/processor.rs

```diff
@@ -246,20 +246,36 @@ fn process_sell_rewards(program_id: &Pubkey, accounts_raw: &[AccountInfo]) -> Pr
         spl_token::state::Mint::unpack_from_slice(&accounts.b_sol_mint.data.borrow())?;
     let b_sol_supply = token_mint_state.supply;
 
-    let reserve_st_sol = StLamports(Anker::get_token_amount(accounts.st_sol_reserve_account)?);
+    let reserve_st_sol_before =
+        StLamports(Anker::get_token_amount(accounts.st_sol_reserve_account)?);
 
     // Get StLamports corresponding to the amount of b_sol minted.
     let st_sol_amount = solido.exchange_rate.exchange_sol(Lamports(b_sol_supply))?;
 
     // If `reserve_st_sol` < `st_sol_amount` something went wrong, and we abort the transaction.
-    let rewards = (reserve_st_sol - st_sol_amount)?;
+    let rewards = (reserve_st_sol_before - st_sol_amount)?;
 
     // Get the amount of UST that we had.
     let ust_before = MicroUst(Anker::get_token_amount(accounts.ust_reserve_account)?);
     swap_rewards(program_id, rewards, &anker, &accounts)?;
     // Get new UST amount.
     let ust_after = MicroUst(Anker::get_token_amount(accounts.ust_reserve_account)?);
+    let reserve_st_sol_after =
+        StLamports(Anker::get_token_amount(accounts.st_sol_reserve_account)?);
     let swapped_ust = (ust_after - ust_before)?;
+    let swapped_st_sol = (reserve_st_sol_before - reserve_st_sol_after)?;
+
+    // The token swap program should not take more stSOL than we told it to swap.
+    // As an extra line of defense, confirm this after the swap is done, and abort
+    // if some stSOL went missing.
+    if swapped_st_sol > st_sol_amount {
```

Comment:
Thanks for the check, this would have minimized the impact of the bug.

Reply:
This check is incorrect! The swap amount is `rewards`, not `st_sol_amount`!

Reply:
Yes, this is embarrassing. Sorry about that, need to be more careful.